packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -58,6 +59,96 @@ const {
 const { fakeFileFactory, fakeGranuleFactoryV2 } = require('../../../lib/testUtils');
 const Granule = require('../../../models/granules');
 
+/**
+ * Helper function for updating an existing granule with a static payload and validating
+ *
+ * @param {Object} t -- Used for the test context
+ * @param {boolean} writeFromMessage -- Calls writeGranulesFromMessage function if true,
+ *   writeGranuleFromApi otherwise
+ * @returns {Object} -- Updated granule objects from each datastore and PG-translated payload
+ *   updatedPgGranuleFields,
+ *   pgGranule,
+ *   esGranule,
+ *   dynamoGranule,
+ **/
+const updateGranule = async (t, writeFromMessage = false) => {
+  const {
+    collectionCumulusId,
+    esClient,
+    esGranulesClient,
+    executionCumulusId,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    providerCumulusId,
+    knex,
+  } = t.context;
+
+  // Update existing granule with a partial granule object
+  const updateGranulePayload = {
+    granuleId,
+    collectionId: granule.collectionId,
+    cmrLink: 'updatedGranuled.com', // Only field we're changing
+    // FUTURE: In order to update a granule, the payload must include status and
+    // the status must be 'completed' or 'failed'
+    // if it's running or queued, it will try to insert the granule, not upsert
+    status: granule.status,
+  };
+
+  if (writeFromMessage) {
+    const updatedCumulusMessage = {
+      cumulus_meta: {
+        workflow_start_time: t.context.workflowStartTime,
+        state_machine: t.context.stateMachineArn,
+        execution_name: t.context.executionName,
+      },
+      meta: {
+        status: 'completed', // FUTURE: A 'running' state will trigger an insert, not an update
+        collection: t.context.collection,
+        provider: t.context.provider,
+      },
+      payload: {
+        granules: [updateGranulePayload],
+      },
+    };
+
+    await writeGranulesFromMessage({
+      cumulusMessage: updatedCumulusMessage,
+      executionCumulusId,
+      providerCumulusId,
+      knex,
+      granuleModel,
+    });
+  } else {
+    await writeGranuleFromApi({ ...updateGranulePayload }, knex, esClient, 'Update');
+  }
+
+  const dynamoGranule = await granuleModel.get({ granuleId });
+  const pgGranule = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+  const esGranule = await esGranulesClient.get(granuleId);
+
+  // Updates were applied to all datastores
+  t.is(pgGranule.cmr_link, updateGranulePayload.cmrLink);
```

Comment:
Oh, I see, this is just ensuring the updates are made ... so it can fail here?  

I'm wondering if these evaluations should be test specific given we should be doing full record eval if that's what the test is validating?     Is this intended to test the test?

Reply:
Yeah, so this check in the `updateGranule` helper is intended to verify that the update happened. The idea was to have a shallow "Nothing went wrong with the initial `writeGranuleFrom...`" call. The actual unit tests do the work we're more interested in:

Comparing the update payload to what was actually written in PG:
https://github.com/nasa/cumulus/pull/3082/files/9794f966250a13dee5704a469f3aa48c65487646#diff-e4b95ee0e3b92433a8e670895c1e8da32d083c3ca64e31761d4bb353cbb791a7R1585-R1590

And comparing what was written in PG to Dynamo and Elastic:
https://github.com/nasa/cumulus/pull/3082/files/9794f966250a13dee5704a469f3aa48c65487646#diff-e4b95ee0e3b92433a8e670895c1e8da32d083c3ca64e31761d4bb353cbb791a7R1615-R1626